src/components/CreateRecommendationModal/index.tsx

```diff
@@ -0,0 +1,318 @@
+    const [selectedLeague, setSelectedLeague] = useState<number | undefined>();
+
+    // Fetch countries
+    const { data: countries = [] } = useQuery({
+        queryKey: ['countries'],
+        queryFn: async () => {
+            const res = await fetch(`https://v3.football.api-sports.io/countries`, {
+                headers: { 'x-apisports-key': API_KEY },
+            });
+            const data = await res.json();
+            return data.response;
+        },
+        enabled: !!isOpen,
+    });
+
+    // Fetch stadiums and leagues based on country
+    const { data: leagues = [] } = useQuery({
+        queryKey: ['leagues', selectedCountry],
+        queryFn: async () => {
+            if (!selectedCountry) return [];
+            const res = await fetch(`https://v3.football.api-sports.io/leagues?country=${selectedCountry}`, {
+                headers: { 'x-apisports-key': API_KEY },
+            });
+            const data = await res.json();
+            return data.response;
+        },
+        enabled: !!selectedCountry,
+    });
+
+    const { data: stadiums = [] } = useQuery({
+        queryKey: ['stadiums', selectedCountry],
+        queryFn: async () => {
+            if (!selectedCountry) return [];
+            const res = await fetch(`https://v3.football.api-sports.io/venues?country=${selectedCountry}`, {
+                headers: { 'x-apisports-key': API_KEY },
+            });
+            const data = await res.json();
+            return data.response;
+        },
+        enabled: !!selectedCountry,
+    });
+
+    const { data: teams = [] } = useQuery({
+        queryKey: ['teams', selectedLeague],
+        queryFn: async () => {
+            if (!selectedLeague) return [];
+            const res = await fetch(`https://v3.football.api-sports.io/teams?league=${selectedLeague}&season=2023`, {
+                headers: { 'x-apisports-key': API_KEY },
+            });
+            const data = await res.json();
+            return data.response;
+        },
+        enabled: !!selectedLeague,
+    });
+
+    const {
+        control,
+        handleSubmit,
+        setValue,
+        reset,
+        formState: { errors },
+    } = useForm<CreateRecommendationModalValues>({
+        resolver: yupResolver(RecommendationSchema),
+    });
+
+    const onSubmit = async (values: CreateRecommendationModalValues) => {
+        console.log('Submitting recommendation:', values);
+        const { title, description, country, date, stadium, league, homeTeam, awayTeam } = values;
+
+        const match = await MatchService.createMatch({
+            country,
+            stadium,
+            league,
+            date,
+            homeTeam,
+            awayTeam,
+        });
+        await RecommendationService.createRecommendation({
+            title,
+            description,
+            matchId: match!._id,
+            picture: imageUrl, // Store image URL
+            createdBy: '123412341234123412341234',
+        });
```

Comment:
```suggestion
        if (!!match) {
        await RecommendationService.createRecommendation({
            title,
            description,
            matchId: match._id,
            picture: imageUrl, // Store image URL
            createdBy: '123412341234123412341234',
        });
        }
```
dont use ! operator in code but rather check the response to see if it is not undefined